crates/rolldown/src/ast_scanner/import_assign_analyzer.rs

```diff
@@ -0,0 +1,53 @@
+use oxc::ast::{
+  ast::{IdentifierReference, UnaryOperator},
+  AstKind,
+};
+use rolldown_error::BuildDiagnostic;
+
+use super::AstScanner;
+
+impl<'me, 'ast: 'me> AstScanner<'me, 'ast> {
+  pub fn check_import_assign(&mut self, ident: &IdentifierReference) {
+    if let Some(potential_update) =
+      self.potential_update_import_reference.get(&ident.reference_id())
+    {
+      match potential_update {
+        super::PotentialImportUpdateReference::Namespace => {
+          // TODO: namespace detection
```

Comment:
Needs some refactor, assign to import with **namespace** is a little complex that needs some refactor, so I left in another pr.